scripts/BBoxManage.sh

```diff
@@ -0,0 +1,474 @@
+    fi
+
+    cp $PASSWORD_CONFIG.original $PASSWORD_CONFIG 2>&1 | logger -t $(basename $0)
+
+    if [ ${PIPESTATUS[0]} -ne 0 ]
+    then
+      failure
+    else
+      reload_nginx
+    fi
+  fi
+}
+
+function reset () {
+  restore_password_config
+  restore_hostapd_config
+  restore_nginx_config
+}
+
+function success () {
+  echo $SUCCESS
+  exit 0;
+}
+
+function failure () {
+  echo $FAILURE
+  exit 1;
+}
+
+function reload_nginx () {
+  # Restart the nginx service
+  if [ $DEBUG == 1 ]; then
+    echo "Reloading nginx configuration"
+  fi
+
+  /etc/init.d/nginx reload
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function unmountusb () {
+  # Unmount usb
+  if [ $DEBUG == 1 ]; then
+    echo "Unmounting USB media"
+  fi
+
+  pumount /media/usb0 2>&1 | logger -t $(basename $0)
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function doshutdown () {
+  # Shutdown
+  if [ $DEBUG == 1 ]; then
+    echo "Shutting down"
+  fi
+
+  /sbin/shutdown now 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function doreboot () {
+  # Reboot
+  if [ $DEBUG == 1 ]; then
+    echo "Rebooting"
+  fi
+
+  /sbin/reboot 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function restart_hostapd () {
+  # Restart the hostapd service
+  if [ $DEBUG == 1 ]; then
+    echo "Restarting hostapd"
+  fi
+
+  /etc/init.d/hostapd restart
+
+  if [ $? -eq 0 ]
+  then
+    success
+  else
+    failure
+  fi
+}
+
+function set_password () {
+  if [[ -z "${val// }" ]]; then
+    echo 'Missing password value'
+    exit 1;
+  fi
+
+  backup_password_config
+
+  # Update the password property in the basicauth configuration file
+  if [ $DEBUG == 1 ]; then
+    echo "Updating password to '$val'"
+  fi
+
+  local new_hash=`echo $val | openssl passwd -apr1 -salt $PASSWORD_SALT -stdin`
+  echo "biblebox:$new_hash" > $PASSWORD_CONFIG 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    reload_nginx
+  else
+    failure
+  fi
+}
+
+function get_hostname () {
+  local hostname=`grep 'bbox_server_name=' $NGINX_CONFIG`
+  local idx=`expr index "$hostname" =`
+
+  web_hostname="${hostname:$idx:${#hostname}-$idx}"
+
+  echo $web_hostname;
+}
+
+function set_hostname () {
+  if [[ -z "${val// }" ]]; then
+    echo 'Missing hostname value'
+    exit 1;
+  fi
+
+  backup_nginx_config
+
+  # Update the hostname in the nginx config
+  if [ $DEBUG == 1 ]; then
+    echo "Updating hostname to '$val'"
+  fi
+
+  get_hostname > /dev/null #suppress output
+
+  sed -i "s/$web_hostname/$val/g" $NGINX_CONFIG 2>&1 | logger -t $(basename $0)
+
+  if [ ${PIPESTATUS[0]} -eq 0 ]
+  then
+    reload_nginx
+  else
+    failure
+  fi
+}
+
+function get_channel () {
+  local channel=`grep '^channel=' $HOSTAPD_CONFIG`
+  if [[ $channel == channel=\"* ]]; then
```

Comment:
So looking further, the valid channels are numbers and the string "default" (which has the same meaning as 0). If we want to remove the if statement, it should be safe to do so given we control the input (and specify a numeric value in our playbook)